lesson-10/dobricsongor/Polygons.py

```diff
@@ -0,0 +1,81 @@
+class Polygon(object):
+
+    def __init__(self, *args):
+        self.sides = args
+
+    def __str__(self):
+        return f'Polygon has {len(self.sides)} sides'
+
+    def display(self):
+        for side_index, length in enumerate(self.sides, start=1):
+            print('Side {} with length: {}'.format(side_index, length))
+
+
+class Triangle(Polygon):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        s1, s2, s3 = self.sides
+        s_p = sum(self.sides) / 2
+        return f'The area of this triangle is {round(((s_p*(s_p - s1)*(s_p - s2)*(s_p - s3))** 0.5),2)}.'
+
+class PerimeterMixin:
+    
+    def perimeter(self):
+        return f'The perimeter of this shape is {sum(self.sides)}.'
```

Comment:
genu asta de metode ii bine sa returneze valoarea in sine, nu un string. Partea de display poate fi adaugata intr-o metoda separata